slack-base/src/main/java/com/hubspot/slack/client/models/dialog/form/elements/helper/SlackDialogElementNormalizer.java

```diff
@@ -0,0 +1,152 @@
+import com.hubspot.slack.client.models.dialog.form.elements.AbstractSlackFormTextElement;
+import com.hubspot.slack.client.models.dialog.form.elements.AbstractSlackFormTextareaElement;
+import com.hubspot.slack.client.models.dialog.form.elements.HasLabel;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackDialogFormElement;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackDialogFormElementLengthLimits;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormOption;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormOptionGroup;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormOptionGroupIF;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormOptionIF;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormSelectElement;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormTextElement;
+import com.hubspot.slack.client.models.dialog.form.elements.SlackFormTextareaElement;
+
+public class SlackDialogElementNormalizer {
+  private SlackDialogElementNormalizer() {
+  }
+
+  public static AbstractSlackFormTextElement normalize(AbstractSlackFormTextElement element) {
+    if (shouldNormalizePlaceholder(element, SlackDialogFormElementLengthLimits.MAX_PLACEHOLDER_LENGTH)
+        || shouldNormalizeLabel(element, SlackDialogFormElementLengthLimits.MAX_LABEL_LENGTH)
+        || shouldNormalize(element.getHint(), SlackDialogFormElementLengthLimits.MAX_HINT_LENGTH)) {
+      return SlackFormTextElement.copyOf(element)
+          .withPlaceholder(normalizePlaceholder(element))
+          .withLabel(normalizeLabel(element))
+          .withHint(normalizeHint(element));
+    }
+    return element;
+  }
+
+  public static AbstractSlackFormTextareaElement normalize(AbstractSlackFormTextareaElement element) {
+    if (shouldNormalizePlaceholder(element, SlackDialogFormElementLengthLimits.MAX_PLACEHOLDER_LENGTH)
+        || shouldNormalizeLabel(element, SlackDialogFormElementLengthLimits.MAX_LABEL_LENGTH)
+        || shouldNormalize(element.getHint(), SlackDialogFormElementLengthLimits.MAX_HINT_LENGTH)) {
+      return SlackFormTextareaElement.copyOf(element)
+          .withPlaceholder(normalizePlaceholder(element))
+          .withLabel(normalizeLabel(element))
+          .withHint(normalizeHint(element));
+    }
+    return element;
+  }
+
+  public static AbstractSlackFormSelectElement normalize(AbstractSlackFormSelectElement element) {
+    if (shouldNormalizePlaceholder(element, SlackDialogFormElementLengthLimits.MAX_PLACEHOLDER_LENGTH)
+        || shouldNormalizeLabel(element, SlackDialogFormElementLengthLimits.MAX_LABEL_LENGTH)
+        || shouldNormalize(element.getOptionGroups(), SlackDialogFormElementLengthLimits.MAX_OPTION_GROUPS_NUMBER)
+        || shouldNormalize(element.getOptions(), SlackDialogFormElementLengthLimits.MAX_OPTIONS_NUMBER)) {
```

Comment:
The same is duplicated on line 73, so we can extract it to a separate method. But I don't insist on this chanhe.